CriticalMass/ChatViewController.swift

```diff
@@ -7,24 +7,32 @@
 
 import UIKit
 
-class ChatViewController: UIViewController, ChatInputDelegate {
+protocol ChatInputDelegate: AnyObject {
+    func didTapSendButton(text: String, completionHandler: ChatViewController.CompletionHandler?)
+}
+
+class ChatViewController: UIViewController {
+    typealias CompletionHandler = (Bool) -> Void
     private enum Constants {
-        static let chatInputHeight: CGFloat = 64
+        static let chatInputHeight: CGFloat = 180
     }
-    private let chatInput = ChatInputView(frame: .zero)
+
     private let messagesTableViewController = MessagesTableViewController<ChatMessageTableViewCell>(style: .plain)
     private let chatManager: ChatManager
+    private let chatInput: ChatInputViewController
     private lazy var chatInputBottomConstraint = {
-        NSLayoutConstraint(item: chatInput, attribute: .bottom, relatedBy: .equal, toItem: view, attribute: .bottom, multiplier: 1, constant: 0)
+        NSLayoutConstraint(item: chatInput.view!, attribute: .bottom, relatedBy: .equal, toItem: view, attribute: .bottom, multiplier: 1, constant: 0)
     }()
 
     private lazy var chatInputHeightConstraint = {
-        chatInput.heightAnchor.constraint(equalToConstant: Constants.chatInputHeight)
+        NSLayoutConstraint(item: chatInput.view!, attribute: .height, relatedBy: .lessThanOrEqual, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: Constants.chatInputHeight)
     }()
 
-    init(chatManager: ChatManager) {
+    init(chatManager: ChatManager, chatInputViewController: ChatInputViewController) {
```

Comment:
What are the benefits here passing in the ChatInputViewController with the initialiser? Maybe create it within the ChatViewController?

Reply:
I did before but the input requires the themeController now since I can not set the keyboard appearance via uiappearance for the textView and since we injected that before I tought I should do the same here for consitancy. TBH the themeController approach seems not very practical anymore but maybe I can do a refactor PR next

Reply:
We could inject the ThemeController for now into the ChatViewController to create the ChatInputViewController inside the ChatViewController to keep it out of the AppController

Reply:
Fine with me. What if I say I am thinking  a themController Singleton? Think for testing it should still do the job

Reply:
It might to the job right now but it might become painful in the future. For now I would prefer injecting the theme controller into the view controller that need it